src/routes/home/Home.js

```diff
@@ -124,31 +124,43 @@ function extractLocationDateFromVideo({ attachmentArrayBuffer }) {
 }
 
 async function extractLocation({ attachmentFile, attachmentArrayBuffer, ext }) {
-  if (isVideo({ ext })) {
-    return extractLocationDateFromVideo({ attachmentArrayBuffer })[0];
-  }
-  if (!isImage({ ext })) {
-    throw new Error(`${attachmentFile.name} is not an image/video`);
-  }
+  try {
+    if (isVideo({ ext })) {
+      return extractLocationDateFromVideo({ attachmentArrayBuffer })[0];
+    }
+    if (!isImage({ ext })) {
+      throw new Error(`${attachmentFile.name} is not an image/video`);
+    }
+
+    const { latitude, longitude } = await exifr.gps(attachmentArrayBuffer);
 
-  const { latitude, longitude } = await exifr.gps(attachmentArrayBuffer);
+    return { latitude, longitude };
+  } catch (err) {
+    console.error(err);
 
-  return { latitude, longitude };
+    throw 'location'; // eslint-disable-line no-throw-literal
+  }
 }
 
 async function extractDate({ attachmentFile, attachmentArrayBuffer, ext }) {
-  if (isVideo({ ext })) {
-    return extractLocationDateFromVideo({ attachmentArrayBuffer })[1];
-  }
-  if (!isImage({ ext })) {
-    throw new Error(`${attachmentFile.name} is not an image/video`);
-  }
+  try {
+    if (isVideo({ ext })) {
+      return extractLocationDateFromVideo({ attachmentArrayBuffer })[1];
+    }
+    if (!isImage({ ext })) {
+      throw new Error(`${attachmentFile.name} is not an image/video`);
+    }
+
+    const { CreateDate } = await exifr.parse(attachmentArrayBuffer, [
+      'CreateDate',
+    ]);
 
-  const { CreateDate } = await exifr.parse(attachmentArrayBuffer, [
-    'CreateDate',
-  ]);
+    return CreateDate.getTime();
+  } catch (err) {
+    console.error(err);
 
-  return CreateDate.getTime();
+    throw 'creation date'; // eslint-disable-line no-throw-literal
```

Comment:
It's somewhat indirect to be throwing these values in these three functions, maybe move them into the `Promise.allSettled` call below?

Reply:
72d6e7a addresses this, but it puts a weird `RangeError: Invalid time value` in the error modal
for DSC_0605.JPG though

![image](https://user-images.githubusercontent.com/6473925/110135438-85514c80-7d9c-11eb-8a5b-930d52511d7d.png)